Userland/Libraries/LibJS/Runtime/Date.cpp

```diff
@@ -120,6 +125,171 @@ static constexpr double MS_PER_HOUR = 3600000;
 // https://tc39.es/ecma262/#eqn-msPerDay
 static constexpr double MS_PER_DAY = 86400000;
 
+// DayWithinYear(t), https://tc39.es/ecma262/#eqn-DayWithinYear
+u16 day_within_year(double t)
+{
+    // Day(t) - DayFromYear(YearFromTime(t))
+    return static_cast<u16>(day(t) - day_from_year(year_from_time(t)));
+}
+
+// DateFromTime(t), https://tc39.es/ecma262/#sec-date-number
+u8 date_from_time(double t)
+{
+    switch (month_from_time(t)) {
+    // DayWithinYear(t) + 1𝔽 if MonthFromTime(t) = +0𝔽
+    case 0:
+        return day_within_year(t) + 1;
+    // DayWithinYear(t) - 30𝔽 if MonthFromTime(t) = 1𝔽
+    case 1:
+        return day_within_year(t) - 30;
+    // DayWithinYear(t) - 58𝔽 - InLeapYear(t) if MonthFromTime(t) = 2𝔽
+    case 2:
+        return day_within_year(t) - 58;
```

Comment:
The leap year is not yet accounted for in these cases I guess.

Reply:
Good catch, thanks!